modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbarAnchor.ts

```diff
@@ -62,9 +69,12 @@ const determineInsideLayout = (editor: Editor, contextbar: SugarElement<HTMLElem
     // preserve or show at the top for a new anchor element.
     return isSameAnchorElement ? LayoutInset.preserve : LayoutInset.north;
   } else if (isSameAnchorElement) {
+    // Preserve the position, get the bounds and then see if we have an overlap.
     // If overlapping and this wasn't triggered by a reposition then flip the placement
-    const isOverlapping = data.lastPos().exists((box) => isVerticalOverlap(selectionBounds, box));
-    return isOverlapping && !data.isReposition() ? LayoutInset.flip : LayoutInset.preserve;
+    return preservePosition(contextbar, () => {
```

Comment:
This changes it so we look up the current position instead of the previous position. What was being done worked fine for now, but doesn't work for transitions since the position doesn't update immediately.